Mathlib/RingTheory/HahnSeries/Lex.lean

```diff
@@ -0,0 +1,169 @@
+/-
+Copyright (c) 2025 Weiyi Wang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Weiyi Wang
+-/
+import Mathlib.Algebra.Order.Group.Abs
+import Mathlib.Algebra.Order.Monoid.Defs
+import Mathlib.Order.PiLex
+import Mathlib.RingTheory.HahnSeries.Addition
+
+/-!
+
+# Lexicographical order on Hahn series
+
+In this file, we define lexicographical ordered `Lex (HahnSeries Γ R)`, and show this
+is a `LinearOrder` when `Γ` and `R` themselves are linearly ordered. Additionally,
+it is an ordered group when `R` is.
+
+-/
+
+namespace HahnSeries
+
+section PartialOrder
+
+variable {Γ : Type*} {R : Type*}
```

Comment:
Can you move this to the outer namespace, and remove the duplicate `Type` variable declarations?